app/internal_packages/preferences/lib/tabs/workspace-section.tsx

```diff
@@ -92,12 +94,13 @@ class LaunchSystemStartItem extends React.Component {
 
     const available = await service.checkAvailability();
     if (!this._mounted) return;
-    this.setState({ available });
 
     if (available) {
-      const launchOnStart = service.doesLaunchOnSystemStart();
+      const launchOnStart = await service.doesLaunchOnSystemStart();
```

Comment:
If I understand the code correctly, I think that only this line of code change is needed in order to achieve the expected result.

In my opinion, the code with two boolean flags (as original) is easier to understand than mapping the question whether the configuration field is displayed at all into the state of the option with the `unavailable` string.

Reply:
You are correct that is the only line that *needs* to change, but I wanted to make it consistent. If I were to change from the bool-or-string pattern, I'd want to do that to the Default Mail Client option as well.

@bengotow, what do you think?

Reply:
I've kept thinking about this, and if it were me, I'd actually prefer the bool-or-string technique here, simply because *unavailable, True, and False must be mutually exclusive options*! This completely an erroneous state from occurring.